src/language/runtime/environment.ts

```diff
@@ -355,6 +414,100 @@ export class Environment implements StaticScope {
         if (!this._parent) return land(undefined);
         return jump(() => this._parent!.resolveVarImpl(varname));
     }
+
+    public declareInternalClass<C>(
+        clsName: string,
+        clsContructor: ((args: RuntimeVal[]) => C) | null,
+        clsMethods: Record<string, (o: C, args: RuntimeVal[]) => RuntimeVal>,
+        clsAttributes: Record<string, (o: C) => RuntimeVal>,
+    ) {
```

Comment:
```suggestion
    public declareInternalClass<C>({
        clsName,
        clsConstructor,
        clsMethods = {},
        clsAttributes = {},
    }: {
        clsName: string,
        clsContructor?: (args: RuntimeVal[]) => C,
        clsMethods?: Record<string, (self: C, args: RuntimeVal[]) => RuntimeVal>,
        clsAttributes?: Record<string, (self: C) => RuntimeVal>,
    }) {
```

Formattierung sei dir überlassen. Objekt-destructuring ist auf jeden Fall lesbarer als positional arguments für caller.

Reply:
fair point